redisson/src/main/java/org/redisson/codec/ProtobufCodec.java

```diff
@@ -0,0 +1,210 @@
+package org.redisson.codec;
+
+import com.fasterxml.jackson.databind.ser.BasicSerializerFactory;
+import com.google.protobuf.MessageLite;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.protostuff.LinkedBuffer;
+import io.protostuff.ProtostuffIOUtil;
+import io.protostuff.Schema;
+import io.protostuff.runtime.RuntimeSchema;
+import org.redisson.client.codec.BaseCodec;
+import org.redisson.client.codec.Codec;
+import org.redisson.client.handler.State;
+import org.redisson.client.protocol.Decoder;
+import org.redisson.client.protocol.Encoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ProtobufCodec extends BaseCodec {
+    private static final Logger log = LoggerFactory.getLogger(ProtobufCodec.class);
+    private final Class<?> mapKeyClass;
+    private final Class<?> mapValueClass;
+    private final Class<?> valueClass;
+
+    //classes in blacklist will not be serialized using protobuf ,but instead will use blacklistCodec
+    private final Set<String> protobufBlacklist;
+    //default value is JsonJacksonCodec
+    private final Codec blacklistCodec;
+
+    private final static Set<String> CLASSES_NOT_SUITABLE_FOR_PROTOBUF = new HashSet<>();
+
+    static {
+        try {
+            Field concreteField = BasicSerializerFactory.class.getDeclaredField("_concrete");
```

Comment:
You need to avoid reflection here and extend BasicSerializerFactory class. Then you can add getters for these fields.

Reply:
Done